connect/src/com/telenor/connect/sms/SmsPinParseUtil.java

```diff
@@ -7,6 +7,8 @@
 
 public class SmsPinParseUtil {
 
+    private static final Pattern PIN_PATTERN = Pattern.compile("<#>.*([0-9]{4}).*.{11}");
```

Comment:
The "4" here is the length of the pin? So if we change this we must change the pattern?

Reply:
Yes, yes, and yes.

Reply:
I'm a bit worried that matching the hash with just plain '.' might be too permissive. As it is a base64 fragment, the following should work
```
[A-Za-z0-9\+\/]{11}
````
(if I quoted correctly...)